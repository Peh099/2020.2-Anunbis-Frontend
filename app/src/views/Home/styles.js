import styled from 'styled-components';
import Button from '../../components/Button';

export const Container = styled.div`
    height: 100vh;
    background-image: linear-gradient(#26425F, #1D2935); 
    display: flex;
    flex-direction: column;
    *{
    scroll-behavior: smooth !important;
    }
`

export const Main = styled.main`
    overflow-y: scroll;
    -ms-overflow-style: none; 
    ::-webkit-scrollbar {
    display: none;
    }
    height: 200vh;
    width: 100vw;
    align-items: center;
    align-self: center;
`

export const Footer = styled.main`
background: #212121;
color: #FFFFFF;
width: 100%;
height: 40px;
margin-top: auto;

    p {
        font-family: 'Roboto';
        font-style: normal;
        font-weight: bold;
        font-size: 20px;
        line-height: 40px;
        margin-left: 25px;
    }
`

export const Middle = styled.div`
display: flex;
justify-content: center;
flex-direction: row;
align-items: center;
margin-left: auto;

    div{
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        width: 50vw;
        align-items: center;
    }

    a{
        color: var(--lightWhite);
        display: block;
        padding: 10px;
        text-decoration: ${props => props.isSelected ? 'underline' : 'none'};
        font-size: 24px;
    }
    a:visited{
        text-decoration: underline;
    }
    a:hover{
        text-decoration: underline;
    }
    
    
`;

export const BtnEdition = styled(Button)`
    margin: ${props => props.margin ? props.margin : "0px"};
    width: ${props => props.width ? props.width : "80px"};
    padding: 5px;
    font-size: 14px;
    font-style: 'Roboto';
    letter-spacing: 1.5px;
    border: 0px;
    margin-left: auto;
    color: ${props => props.color ? props.color : "var(--black)"};
    background-color: ${props => props.background ? props.background : "var(--yellow)"};
`

export const Content = styled.div`
    display: ${props => props.display ? props.display : ''};
    flex-direction:row;
    align-items: center;
    justify-content: center;
    width: ${props => props.width ? props.width : '100vw'};
    height: ${props => props.height ? props.height : 'calc(100vh - 40px)'};
    background-color: ${props => props.backColor ? props.backColor : ''};

    text-align: ${props => props.txtAlign ? props.txtAlign : ''};
`;

export const Image = styled.img`
    width: ${props => props.width ? props.width : '60%'};
    height: ${props => props.height ? props.height : '100%'};
    margin: ${props => props.margin ? props.margin : 'auto'};
    border-radius: 0px;
`;

export const Text = styled.div`
    width: ${props => props.width ? props.width : '40%'};
    height: fit-content;
    font-family: 'Roboto';
    font-style: 'normal';
    text-align: center;
    margin-inline-start: 40px;
    label{
        font-size: 40px;
        color: var(--lightWhite);       
    }

    p{
    margin-top: 5px;
    color: #C4C4C4;
    font-size: 24px;
    line-height: 26px;
    text-align: justify;
    justify-content: baseline;
    }
    
`;