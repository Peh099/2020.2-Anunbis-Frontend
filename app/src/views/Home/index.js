import React from 'react';
import { useHistory } from 'react-router';
import { End } from '../../components/LayoutApp/styles';
import Menu from '../../components/Menu';
import { Container, Middle, BtnEdition} from './styles';


export default function Home() {

    const history = useHistory();

    return (
        <Container>
            <Menu background="var(--transparent)">
                <Middle>
                    <div>
                        <a href='/'>Inicio</a>
                        <a href='/'>Equipe</a>
                        <a href='/'>Contato</a>

                    </div>
                </Middle>
                <End height='22,5px'>
                    <BtnEdition text='Login' onClick={() => history.push('user/login')} />
                </End>
            </Menu>
        </Container>
    );
}