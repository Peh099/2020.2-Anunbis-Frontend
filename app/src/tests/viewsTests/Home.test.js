import renderer from 'react-test-renderer';
import Home from '../../views/Home';

describe('Snapshot Home view', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
