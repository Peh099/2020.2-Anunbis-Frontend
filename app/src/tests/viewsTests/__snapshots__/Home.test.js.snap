// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Home view matches the snapshot 1`] = `
<div
  className="sc-pNWdM fEPZne"
>
  <div
    className="sc-gKAaRy dQFOuk"
  >
    <div
      className="sc-iCoGMd deCerP"
    >
      <img
        alt="logo"
        className="sc-fujyAs kpogkV"
        onClick={[Function]}
        src="Name_Logo.png"
      />
    </div>
    <div
      className="sc-iqAclL frrUS"
    >
      <div>
        <a
          className="sc-crzoAE kqdlXi"
          href="#begin"
        >
          Inicio
        </a>
        <a
          className="sc-crzoAE bKCzWu"
          href="#howItWorks"
        >
          Como Funciona?
        </a>
        <a
          className="sc-crzoAE bKCzWu"
          href="#team"
        >
          Equipe
        </a>
        <a
          className="sc-crzoAE bKCzWu"
          href="#contact"
        >
          Contato
        </a>
      </div>
    </div>
    <div
      className="sc-jSFjdj kDArBW"
      height="22,5px"
    >
      <button
        className="sc-bdnxRM bSShnC sc-dIsUp jEQvpa"
        onClick={[Function]}
      >
        Login
      </button>
    </div>
  </div>
  <main
    className="sc-jrsJWt jydbDv"
  >
    <span
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <div
        className="sc-bqGGPW lhfGvQ"
        display="flex"
        id="begin"
      >
        <div
          className="sc-fnVZcZ jzcDkO"
        >
          <label>
            Onde a uni√£o faz a for√ßa
          </label>
          <p>
            Alunos trocando experi√™ncias e opini√µes sobre disciplinas e professores para manter a comunidade acad√™mica da Universidade de Brasilia unida e bem informada.
          </p>
          <button
            className="sc-bdnxRM kvmfUC sc-dIsUp bpzxCb"
            onClick={[Function]}
            width="160px"
          >
            Cadastre-se
          </button>
        </div>
        <img
          alt="Logo"
          className="sc-hBMUJo kSUXzT"
          height="80%"
          src="LogoHome.svg"
        />
      </div>
    </span>
    <span
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <div
        className="sc-bqGGPW lhfGvQ"
        display="flex"
        id="howItWorks"
      >
        <div
          className="sc-fnVZcZ dbvFfF"
        >
          <label>
            Como funciona?
          </label>
          <p>
            Os alunos podem escrever avalia√ß√µes p√∫blicas sobre os professores optando por manter ou n√£o o anonimato. Os demais alunos podem ver as avalia√ß√µes, concordar ou discordar. Todas as notas e coment√°rios s√£o de livre acesso para todos os usu√°rios, incluindo professores.
          </p>
        </div>
        <img
          alt="Puzzle"
          className="sc-hBMUJo joGCGI"
          height="80%"
          src="Puzzle.svg"
        />
      </div>
    </span>
    <span
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <div
        className="sc-bqGGPW jRBYeI"
        direction="column"
      >
        <div
          className="sc-bqGGPW EtnDg"
          display="flex"
          height="50%"
          id="team"
        >
          <div
            className="sc-fnVZcZ dbvFfF"
          >
            <label>
              Equipe
            </label>
            <p>
              O Anunbis √© um projeto criado e desenvolvido por alunos de M√©todos de Desenvolvimento de Software (MDS) da Faculdade do Gama (FGA), da Universidade de Bras√≠lia (UnB).
            </p>
          </div>
          <img
            alt="Team"
            className="sc-hBMUJo fakbmF"
            height="75%"
            src="Teamwork.svg"
          />
        </div>
        <div
          className="sc-bqGGPW gWaFud"
          display="flex"
          height="100px"
        >
          <div
            className="sc-fnVZcZ dbvFfF"
          >
            <label>
              Membros
            </label>
          </div>
        </div>
        <div
          className="sc-ksluID esVTvE"
        >
          <div
            className="sc-fFSPTT kUCAqb"
          >
            <a
              href="https://github.com/RcleydsonR"
              rel="noreferrer"
              target="_blank"
            >
              <div>
                Rafael Cleydson
              </div>
              <img
                alt="Rafael Cleydson"
                src="Rafael.jpeg"
              />
            </a>
          </div>
          <div
            className="sc-fFSPTT kUCAqb"
          >
            <a
              href="https://github.com/Balbinoo"
              rel="noreferrer"
              target="_blank"
            >
              <div>
                Rodrigo Balbino
              </div>
              <img
                alt="Rodrigo Balbino"
                src="Rodrigo.jpeg"
              />
            </a>
          </div>
          <div
            className="sc-fFSPTT kUCAqb"
          >
            <a
              href="https://github.com/mangabeiras"
              rel="noreferrer"
              target="_blank"
            >
              <div>
                Roberto Santana
              </div>
              <img
                alt="Roberto Santana"
                src="Roberto.jpeg"
              />
            </a>
          </div>
          <div
            className="sc-fFSPTT kUCAqb"
          >
            <a
              href="https://github.com/oEduardoAfonso"
              rel="noreferrer"
              target="_blank"
            >
              <div>
                Eduardo Afonso
              </div>
              <img
                alt="Eduardo Afonso"
                src="Eduardo.jpeg"
              />
            </a>
          </div>
          <div
            className="sc-fFSPTT kUCAqb"
          >
            <a
              href="https://github.com/thiagohdaqw"
              rel="noreferrer"
              target="_blank"
            >
              <div>
                Thiago Paiva
              </div>
              <img
                alt="Thiago Paiva"
                src="Thiago.jpeg"
              />
            </a>
          </div>
          <div
            className="sc-fFSPTT kUCAqb"
          >
            <a
              href="https://github.com/victorhugo21"
              rel="noreferrer"
              target="_blank"
            >
              <div>
                Victor Hugo
              </div>
              <img
                alt="Victor Hugo"
                src="Victor.jpeg"
              />
            </a>
          </div>
        </div>
      </div>
    </span>
    <span
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <div
        className="sc-bqGGPW lhfGvQ"
        display="flex"
        id="contact"
      >
        <div
          className="sc-fnVZcZ dbvFfF"
        >
          <label>
            Contate-nos
          </label>
          <p>
            Deseja elogiar, sugerir mudan√ßas, ou contribuir? Entre em contato conosco atrav√©s do email ou acesse nosso reposit√≥rio no github.
          </p>
          <a
            href="mailto:anunbis.team@gmail.com"
          >
            anunbis.team@gmail.com
          </a>
          <a
            href="https://github.com/fga-eps-mds/2020.2-Anunbis"
            rel="noreferrer"
            target="_blank"
          >
            Reposit√≥rio Back-end
          </a>
          <a
            href="https://github.com/fga-eps-mds/2020.2-Anunbis-FrontEnd"
            rel="noreferrer"
            target="_blank"
          >
            Reposit√≥rio Front-end
          </a>
        </div>
        <img
          alt="Img"
          className="sc-hBMUJo dczRep"
          height="50%"
          src="Communication.svg"
        />
      </div>
    </span>
    <main
      className="sc-kEqXSa dveiBy"
    >
      <p>
        üÑØ Anunbis 2021
      </p>
    </main>
  </main>
</div>
`;
